NewForce-Cohort-7/tabloidfullstack-agnostic-pillows
# Ticket #25 - Deactivate User Profiles

# Description
This is on the `gs-deactivateUserProfiles` branch.

For your reference: [SQL Data](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL)

**#25 Deactivate User Profiles** in Trello - [#25 Deactivate User Profiles](https://trello.com/c/QOaYLq27)
Per the ticket: "As an admin I would like the ability to deactivate a User Profile so that I can prevent unsavory characters from using the system.

Given an admin is vising the list of User Profiles
When they select the Deactivate option
Then they should be prompted to confirm deactivation

Given the admin wishes to confirm the deactivation
When they the select the option to confirm
Then the User Profile should be deactivated in the system
And the admin should be directed back to the User Profile list

Given the admin decides not to confirm the deactivation
When they the select the option to reject confirmation
Then the User Profile should NOT be deactivated
And the admin should be directed back to the User Profile list

Given a user is deactivated
When they try to login to the system
Then the system should behave as if the user does not exist"

This functionality has now been implemented. 

## Type of change

# Testing Instructions

**GIT this branch:**
1. Clone this repository to your own machine using `git clone https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows.git`
2. `git fetch -a` in the master branch to retrieve this branch named `gs-deactivateUserProfiles`.
3.  checkout this branch using the command `git checkout gs-deactivateUserProfiles`


**Create the database:**
1. Open Visual Studio.
2. At the start screen select "Open a Local Folder"
4. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, then select the SQL folder
5. Run all three .sql files in the [SQL folder](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL) (The first file creates the database and tables if they don’t already exist, and the second file inserts a very small amount of sample data into those tables for testing. The third SQL script inserts the new column for the `IsActive` boolean necessary for account deactivation and reactivation).

**Run the app:**
1. Open Visual Studio
2. At the start screen select Open a project or solution.
3. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, and select the .sln file to run the solution.
4. Hit the Green play button on the toolbar that says TabloidFullStack.
5. This will launch the Swagger page in your default browser. The Swagger page provides the endpoints and data from the SQL server that we created. 
6. Now we need to run the React side of the app. 
7. In your command line, cd into the `client` folder and into the `tabloid` folder. 
8. Start by running `npm install react-scripts`. Once this has been installed, type `npm start`. 
9. This will open the React app in your default browser. You should now see the log-in page. 
10. Log in to user `foo@bar.com`. You do not need to enter a password. 

**Test the new feature:**
1. Click on 'Users' on the navigation bar.
2. You will be brought to a User Profile list page. 
3. You should see cards that include the user's full name, display name, and user type. 
4. As an admin, you should also be able to see the following:
5. Account status and a red deactivation button. The cards will be sorted by active to deactivated status (active first, deactivated second).
6. Click on the deactivate button for the user **Tobi Figiovanni**.
7. This will prompt you to deactivate the account. If you hit cancel, nothing will happen. If you hit accept, you will deactivate that user and also have the user list updated. You should now see Tobi as deactivated in the list.
8. Hit log out on the navigation bar. Try to log in as Tobi (tfigiovanni5@baidu.comx). You should now receive an error message that the account is invalid or has been deactivated. 

# Checklist:
